src-v2/NTS.Domain.Setup/Entities/Loop.cs

```diff
@@ -1,19 +1,63 @@
 ﻿using NTS.Domain.Setup.Import;
+using Newtonsoft.Json;
 
 namespace NTS.Domain.Setup.Entities;
 
-public class Loop : DomainEntity, ISummarizable, IImportable
+public class Loop : DomainEntity, ISummarizable
 {
+    public static Loop Create(double distance, int rec, int rest, bool isFinal) => new(distance, rec, rest, isFinal);
+
+    public static Loop Update(int id, double distance, int rec, int rest, bool isFinal) => new(id, distance, rec, rest, isFinal);
+
+    [JsonConstructor]
+    public Loop(int id, double distance, int rec, int rest, bool isFinal)
+    {
+        if (distance <= 0)
+        {
+            throw new DomainException(nameof(Distance), "Distance cannot be zero or less.");
+        }
+
+        if (rec <= 0)
+        {
+            throw new DomainException(nameof(Recovery), "Recovery time cannot be zero or less.");
+        }
+
+        if (rest <= 0)
+        {
+            throw new DomainException(nameof(Rest), "Rest duration cannot be zero or less.");
+        }
+
+        Id = id;
+        Distance = distance;
+        Recovery = rec;
+        Rest = rest;
+        IsFinal = isFinal;
+    }
     public Loop(double distance, int rec, int rest, bool isFinal)
     {
-		this.Distance = distance;
-		this.Rec = rec;
-		this.Rest = rest;
-		this.IsFinal = isFinal;
+        if (distance <= 0)
+        {
+            throw new DomainException(nameof(Distance), "Distance cannot be zero or less.");
+        }
+
+        if (rec <= 0)
+        {
+            throw new DomainException(nameof(Recovery), "Recovery time cannot be zero or less.");
+        }
+
+        if (rest <= 0)
+        {
+            throw new DomainException(nameof(Rest), "Rest duration cannot be zero or less.");
+        }
+
+        Distance = distance;
+		Recovery = rec;
+		Rest = rest;
+		IsFinal = isFinal;
 	}
 
     public double Distance { get; private set; }
-	public int Rec { get; private set; }
+	public int Recovery { get; private set; }
```

Comment:
Added a semantically complete property name because it makes the code more readable - when one reads it, it gives information about its purpose and avoids a certain level of ambiguity.